src/libs/state/BufferState.js

```diff
@@ -396,6 +404,46 @@ export default class BufferState {
     scrollToMessage(id) {
         this.state.$emit('messagelist.scrollto', { id: id });
     }
+
+    getLoadingState() {
+        const networkState = this.getNetwork().state;
+
+        if (networkState === 'disconnected') {
+            return 'disconnected';
+        } else if (networkState === 'connecting') {
+            return 'connecting';
+        } else if (networkState === 'connected') {
+            if (
+                this.flags.is_requesting_chathistory ||
+                (this.chathistory_request_count === 0 &&
+                    this.joined &&
+                    this.enabled)
+            ) {
+                return 'loading';
+            }
+
+            return 'done';
+        }
+
+        return 'unknown';
+    }
+
+    isReady() {
+        return this.getLoadingState() === 'done';
+    }
+
+    onNetworkConnecting(event) {
+        if (event.network === this.getNetwork()) {
+            this.chathistory_request_count = 0;
+        }
+    }
+
+    onBufferClose(event) {
+        if (event.buffer === this) {
+            this.state.$off('network.connecting', this.onNetworkConnecting);
+            this.state.$off('buffer.close', this.onBufferClose);
```

Comment:
These 2 $off calls won't do anything as the event listeners are attached using `.bind()` which returns a new function, not these specific ones being removed